internal/controlplane/handlers_evalstatus.go

```diff
@@ -17,220 +17,331 @@ package controlplane
 
 import (
 	"context"
-	"time"
+	"fmt"
 
+	"github.com/google/uuid"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 	"google.golang.org/protobuf/types/known/timestamppb"
 
-	"github.com/stacklok/minder/internal/engine"
+	"github.com/stacklok/minder/internal/auth"
+	"github.com/stacklok/minder/internal/db"
 	minderv1 "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
 )
 
-// ListEvaluationResults lists the evaluation results for entities filtered by entity type, labels, profiles, and rule types.
+// ListEvaluationResults lists the evaluation results for entities filtered b
+// entity type, labels, profiles, and rule types.
 func (s *Server) ListEvaluationResults(
 	ctx context.Context,
-	_ *minderv1.ListEvaluationResultsRequest,
+	in *minderv1.ListEvaluationResultsRequest,
 ) (*minderv1.ListEvaluationResultsResponse, error) {
-	entityCtx := engine.EntityFromContext(ctx)
+	userInfo, err := s.store.GetUserBySubject(ctx, auth.GetUserSubjectFromContext(ctx))
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Eerror getting user details: %v", err)
+	}
+
+	if _, err := uuid.Parse(in.GetProfile()); err != nil && in.GetProfile() != "" {
+		return nil, status.Error(codes.InvalidArgument, "Error invalid profile ID")
+	}
 
-	err := entityCtx.Validate(ctx, s.store)
+	// Read the list if projects the user has access to
+	projects, err := s.authzClient.ProjectsForUser(ctx, userInfo.IdentitySubject)
```

Comment:
This part throws me off, normally, we want to encompass a certain project or project tree for a certain operation. This diverges away from that and does a request for all projects a user has access to. While this is a possible way to go... I personally think we should keep operations project-scoped, as that's what we've done for our authorization logic, and it is our tenancy model. Let's use the project from the entity context.

If we need an endpoint for all projects for a user, let that be a new API call with the user scope.

Reply:
OK, I'm fine restricting it. Any objections to not listing all projects @rdimitrov ? We discussed returning everything, but I also think the function should behave as the others.

Reply:
at the moment the only FE consuming this is project scoped. this endpoint is already returning a lot of data anyways, so probably the future improvements lie in somehow restricting the amount of data it returns rather than expanding it even further